Signup: Allow theme slug as a default dependency in plans step

daledupreez opened the pull request:
#### Proposed Changes

* This PR adds support for the plans step to specify `themeSlugWithRepo` as a default dependency, which is accomplished by adding some logic to the `isPlanFulfilled()` function to add `defaultDependencies?.themeSlugWithRepo` to the submitted step data if the value is present
* This PR also specifies `themeSlugWithRepo` to be `pub/twentytwentytwo` for the `ecommerce` and `ecommerce-monthly` flows, where the plans step is not actually shown to users, so the changes from https://github.com/Automattic/wp-calypso/pull/68411 don't apply, because the `onSelectPlan()` callback is never triggered

#### Testing Instructions

* Open the live testing branch and navigate to `/start/ecommerce` to start the `ecommerce` signup flow (this is normally triggered from the [Ecommerce landing page](https://wordpress.com/ecommerce/) )
* Work through the flow and verify that your site is created with the Twenty Twenty Two theme
* Repeat the test above starting from `/start/ecommerce-monthly`, and verify that you also get a site created with the Twenty Twenty Two theme.
* Navigate to `/start` and work through the main signup flow - verify that there are no changes to any of those steps. (There shouldn't be, as the only steps that have added `themeSlugWithRepo` to `defaultDependencies` are the two I modified!)

#### Pre-merge Checklist

<!--
Complete applicable items on this checklist **before** merging into trunk. Inapplicable items can be left unchecked.

Both the PR author and reviewer are responsible for ensuring the checklist is completed.
-->

- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?

Related to #67562

daledupreez committed:
Signup: Allow theme slug to be specified as a default dependency in the plans step

markbiek commented:
The two ecommerce cases worked fine but I'm getting a type error when I click "start with a free site"

> Uncaught TypeError: right-hand side of 'in' should be an object, got null

mreishus reviewed (approved):
LGTM

Nonblocking: defaultDependencies and defaultedDependencies are easy to mix up, but I don't have a good alternative name for the second variable to suggest :(

daledupreez committed:
Use less bad name for dependencies coming from defaults

daledupreez commented:
I rebased the branch and that addressed both the e2e errors and the issue @markbiek [noted above](https://github.com/Automattic/wp-calypso/pull/68417#issuecomment-1261406983).

I also renamed `defaultedDependencies` to `dependenciesFromDefaults` to make the name less bad. 😬
